lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py

```diff
@@ -0,0 +1,1206 @@
+
+        return record["collect_barcode_kiosk"] or record["return_utm_barcode"] or \
+            record["pre_scan_barcode"] or None
+
+
+    barcode = specimen_barcode()
+
+    if not barcode:
+        LOG.warning("Could not create Specimen Resource due to lack of barcode.")
+        return None, None
+
+    specimen_identifier = create_identifier(
+        system = f"{INTERNAL_SYSTEM}/sample",
+        value = barcode
+    )
+
+    # YYYY-MM-DD HH:MM:SS in REDCap
+    received_time = record['samp_process_date'].split()[0] if record.get('samp_process_date') else None
+
+    note = None
+
+    if record['able_to_test'] == 'no':
+        note = 'never-tested'
+    else:
+        note = 'can-test'
+
+    specimen_type = 'NSECR'  # Nasal swab.
+    specimen_resource = create_specimen_resource(
+        [specimen_identifier], patient_reference, specimen_type, received_time,
+        collection_date(record, collection_method), note
+    )
+
+    return create_entry_and_reference(specimen_resource, "Specimen")
+
+
+def collection_date(record: dict, collection_method: CollectionMethod) -> Optional[str]:
+    """
+    Determine sample/specimen collection date from the given REDCap *record*.
+    """
+    if collection_method == CollectionMethod.KIOSK:
+        return record["nasal_swab_q"]
+    elif collection_method == CollectionMethod.SWAB_AND_SEND:
+        return record["date_on_tube"] or record["kit_reg_date"] or record["back_end_scan_date"]
+    else:
+        return None
+
+def combine_multiple_fields(record: Dict[Any, Any], field_prefix: str, field_suffix: str = "") -> Optional[List]:
+        """
+        Handles the combining of multiple fields asking the same question such
+        as country and state traveled.
+        """
+        regex = rf'^{re.escape(field_prefix)}[0-9]+{re.escape(field_suffix)}$'
+        empty_value = ''
+        answered_fields = list(filter(lambda f: filter_fields(f, record[f], regex, empty_value), record))
+
+        if not answered_fields:
+            return None
+
+        return list(map(lambda x: record[x], answered_fields))
+
+
+
+def create_enrollment_questionnaire_response(record: dict, patient_reference: dict,
+                                            encounter_reference: dict) -> Optional[dict]:
+    """
+    Returns a FHIR Questionnaire Response resource entry for the enrollment
+    encounter (i.e. encounter of enrollment into the study)
+    """
+
+    # Do not include `core_age_years` because we calculate the age ourselves in the computed questionnaire.
+    integer_questions = [
+        'weight',
+        'height_total'
+    ]
+
+    string_questions = [
+        'study_area',
+        'text_or_email',
+        'text_or_email_attestation',
+        'campus_location',
+        'affiliation',
+        'student_level',
+        'sea_employee_type',
+        'core_house_members',
+        'core_education',
+        'core_income',
+        'vaccine_method',
```

Comment:
These responses are being ingested as an integer, e.g. "2" in test record ID 61 instead of the expected value

Reply:
Ah, these sound like they are coding that needs to be changed in REDCap.

Reply:
In REDCap vaccine_method has these options:
1, Injection (flu shot)
2, Nasal spray (flu mist)
Thus, I'm ingesting the value as an integer.

Reply:
Ah, but I had it ingested as a string. I made it an integer now.

Reply:
No, we want it to be ingested as a string. This is a REDCap side thing that needs to be fixed.

Reply:
This variable was in the list whose values Misja fixed from integer to string in REDCap today. I'm ingesting this as a string.